qiskit_nature/mappers/second_quantization/logarithmic_mapper.py

```diff
@@ -0,0 +1,231 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The Logarithmic Mapper."""
+
+import operator
+
+from enum import Enum
+from fractions import Fraction
+from functools import reduce
+from typing import List, Union, Tuple
+
+import numpy as np
+
+from qiskit.opflow import PauliSumOp
+from qiskit.quantum_info.operators import SparsePauliOp, Operator
+from qiskit_nature.operators.second_quantization import SpinOp
+from .spin_mapper import SpinMapper
+
+
+class EmbedLocation(Enum):
+    """Embed location type"""
+
+    UPPER = "upper"
+    LOWER = "lower"
+
+
+class LogarithmicMapper(SpinMapper):
+    """A mapper for Logarithmic spin-to-qubit mapping [1].
+    In this local encoding transformation, each individual spin S system is represented via
+    the lowest lying 2S+1 states in a qubit system with the minimal number of qubits needed to
+    represent >= 2S+1 distinct states.
+
+    [1]: Mathis, S. V., Guglielmo, M., & Ivano, T. (2020).
+         Toward scalable simulations of lattice gauge theories on quantum computers.
+         Phys. Rev. D, 102 (9), 094501. 10.1103/PhysRevD.102.094501
+    """
+
+    def __init__(self, padding: float = 1, location: EmbedLocation = EmbedLocation.UPPER) -> None:
```

Comment:
I can't help but wonder, given that this only has two values, whether a boolean might be "better". It would not need to add an extra class/enum type that needs to be documented etc i.e.
```
embed_upper: bool = True
```
Where if False then it does lower.

Reply:
I like the idea of using a boolean for this better than my original Enum suggestion, too :+1: Thanks @woodsp-ibm!

Reply:
I agree with @woodsp-ibm. Somehow, I never thought about using a boolean for 2 choices.